matching-notification-resources/src/main/resources/resources/uicomponents/matchingNotification/matchesTable.js

```diff
@@ -1007,21 +1065,123 @@ var PhenoTips = (function (PhenoTips) {
         }
-        var shortEmail = (emails.length > 0) ? (emails[0].substring(0, 9) + "...") :  "";
+        var shortEmail = (patient.emails.length > 0) ? (patient.emails[0].substring(0, 9) + "...") :  "";
         td += '<div name="notification-email-short">' + shortEmail + '</div>';
 
         //if logged as admin - add notification checkbox for local PC patient email contact but not for self (not for patients owned by admin)
-        if (this._isAdmin && serverId == '' && emails.length > 0) {
-            td += '<span class="fa fa-envelope" title="' + this._NOTIFY + '"></span> <input type="checkbox" class="notify" data-matchid="' + matchId + '" data-patientid="'+ patientId +'" data-emails="'+ emails.toString() +'">';
+        if (this._isAdmin && patient.serverId == '' && patient.emails.length > 0) {
+            td += '<span class="fa fa-envelope" title="' + this._NOTIFY + '"></span> <input type="checkbox" class="notify" data-matchid="' + matchId + '" data-patientid="'+ patient.patientId +'" data-emails="'+ patient.emails.toString() +'">';
         }
         td += '</td>';
         return td;
     },
 
-    _getContact : function(record)
+    _generateNotificationHistoryTable : function(records)
+    {
+        var table = '<table class="notification-history-table"><tbody>';
+
+        var header = '<tr>';
+        header += '<th scope="col">' + this._NOTIFICATION_HISTORY_DATE + '</th>';
+        header += '<th scope="col">' + this._NOTIFICATION_HISTORY_FROM + '</th>';
+        header += '<th scope="col">' + this._NOTIFICATION_HISTORY_TO + '</th>';
+        header += '</tr>';
+
+        table += header;
+
+        for (var i=0; i < records.length; i++) {
+            var record = records[i];
+            var row = '<tr class="' + record.type + '" title="' + record.type + '">';
+
+            var date = '<td>';
+            if (record.date) {
+                var dateIconName = record.type == 'contact' ? 'fa fa-envelope-o' : 'fa fa-volume-up' ;
+                date += '<div class="date">' + record.date.split(' ')[0] + '</div><span class="'+ dateIconName + '"> </span></td>';
+            }
+            date += '</td>';
+
+            var from = '<td>';
+            if (record.from) {
+                from += this._generateRecipientCell(record.from, record.type);
+            }
+            from += '</td>';
+
+            var to = '<td>';
+            if (record.to) {
+                to += this._generateRecipientCell(record.to, record.type);
+            }
+            to += '</td>';
+
+            row += date;
+            row += from;
+            row += to;
+            row += '</tr>';
+
+            table += row;
+        }
+
+        table += '</tbody></table>';
+
+        return table;
+    },
+
+    _generateRecipientCell : function(info, type)
+    {
+        // notifications should have something generic e.g. "PhenomeCentral" (no link)
+        if (type == "notification" && info.userinfo.id == "xwiki:XWiki.Admin") {
```

Comment:
>This will not work for Brittney nor any other admin

Right, good catch @sashaandjic . The `From` field should be " PhenomeCentral" for all `notification`s, regardless of which admin sent them.